src/client.cpp

```diff
@@ -789,8 +790,44 @@ void CClient::OnHandledSignal ( int sigNum )
 #endif
 }
 
-void CClient::OnControllerInFaderLevel ( int iChannelIdx, int iValue )
+// Ensure user's own channel is always given MIDI offset 0, so it is the first physical fader
+// All channels with a client ID less than user's own will use ID+1 as their MIDI offset
+
+int CClient::ChanToMIDI ( const int iChannelIdx )
+{
+    if ( iMyChannelID == INVALID_INDEX )
+        return iChannelIdx;
+
+    if ( iChannelIdx == iMyChannelID )
+        return 0;
+
+    if ( iChannelIdx < iMyChannelID )
+        return iChannelIdx + 1;
+
+    return iChannelIdx;
+}
+
+int CClient::MIDIToChan ( const int iMIDIIdx )
+{
+    if ( iMyChannelID == INVALID_INDEX )
+        return iMIDIIdx;
+
+    if ( iMIDIIdx == 0 )
+        return iMyChannelID;
+
+    if ( iMIDIIdx <= iMyChannelID )
+        return iMIDIIdx - 1;
+
+    return iMIDIIdx;
+}
+
+// Handle received MIDI controls
+
+void CClient::OnControllerInFaderLevel ( int iMIDIIdx, int iValue )
 {
+    // map the MIDI index to the real channel number
+    const int iChannelIdx = MIDIToChan ( iMIDIIdx );
```

Comment:
The client shouldn't need to worry about this.  It should have been handled in the MIDI code.